datafusion/sqllogictest/test_files/struct.slt

```diff
@@ -373,3 +373,30 @@ You reached the bottom!
 
 statement ok
 drop view complex_view;
+
+# struct with different keys r1 and r2 is not valid
```

Comment:
Could you also please add a test that shows what happens with coercion when the fields are in a different order?

Like this
```sql
> create table t(a struct<r1 varchar, c int>) as values ({r1: 'foo', c:1}), ({c:2, r: 'bar'});
```

On main it seems to get an error:

> Error during planning: Inconsistent data type across values list at row 1 column 0. Was Struct([Field { name: "r1", data_type: Utf8, nullable: true, dict_id: 0, dict_is_ordered: false, metadata: {} }, Field { name: "c", data_type: Int64, nullable: true, dict_id: 0, dict_is_ordered: false, metadata: {} }]) but found Struct([Field { name: "c", data_type: Int64, nullable: true, dict_id: 0, dict_is_ordered: false, metadata: {} }, Field { name: "r", data_type: Utf8, nullable: true, dict_id: 0, dict_is_ordered: false, metadata: {} }])